pkg/services/olhttp/olrequest.go

```diff
@@ -57,32 +57,33 @@ func (svc OLHTTPService) Read(r interface{}) ([]byte, error) {
 	}
 
 	var (
-		allData []byte
+		allData [][]byte
 		next    string
 	)
+
 	if os.Getenv("OL_LOG_LEVEL") == "debug" {
 		log.Printf("[ONELOGIN HTTP DEBUG] Making Read Request to %s with parameters: %s \n", resourceRequest.URL, resourceRequest.Payload)
 	}
 	resp, data, err := svc.executeHTTP(req, resourceRequest)
 	if err != nil {
-		return []byte{}, err
+		return [][]byte{}, err
 	}
 	for {
-		allData = append(allData, data...)
+		allData = append(allData, data)
 		next = resp.Header.Get("After-Cursor")
 		if next == "" {
 			break
 		}
 		params := req.URL.Query()
-		params.Add("Cursor", next)
+		params.Add("cursor", next)
 		req.URL.RawQuery = params.Encode()
 		resp, data, err = svc.executeHTTP(req, resourceRequest)
 	}
 
 	if err != nil {
-		return []byte{}, err
+		return [][]byte{}, err
 	}
-	return data, err
```

Comment:
This was a very WTF moment for me. This made your tests pass when they should have not.

Reply:
Also, it hid the fact that the byte data appending you were doing was definitely not correct